### PR TITLE
[clang] Add C++26 diagnostics to compatibility diagnosic groups

### PR DESCRIPTION
**Pull request by Endilll**
This patch adds `CXXPre26Compat` and `CXXPre26CompatPedantic` groups (which are concerned with new features not available in older language modes) to `CXX98Compat`, etc. This way, if user has `-Wc++20-compat` and they use pack indexing, they will be warned.

Ideally this should have been done when C++26 groups were created, but we shipped two releases of Clang since then.

### CODE REVIEW

File: clang/include/clang/Basic/DiagnosticGroups.td [hunk 1]
```diff
@@ -384,39 +386,47 @@ def CXX11Compat : DiagGroup<"c++11-compat",
...
 def : DiagGroup<"c++1z-compat", [CXX17Compat]>;
 
-def CXX20Compat : DiagGroup<"c++20-compat", [CXXPre23Compat]>;
+def CXX20Compat : DiagGroup<"c++20-compat", [CXXPre23Compat,
+                                             CXXPre26Compat]>;
 def CXX20CompatPedantic : DiagGroup<"c++20-compat-pedantic",
                                     [CXX20Compat,
-                                     CXXPre23CompatPedantic]>;
+                                     CXXPre23CompatPedantic,
+                                     CXXPre26CompatPedantic]>;
```
Comment (cor3ntin) [inline clang/include/clang/Basic/DiagnosticGroups.td:429]: We don't have a CXX23Compat group?

Comment (Endilll) [reply clang/include/clang/Basic/DiagnosticGroups.td:429]: This is correct.

Comment (cor3ntin) [reply clang/include/clang/Basic/DiagnosticGroups.td:429]: This is weird. We should probably add one for completeness

Comment (Endilll) [reply clang/include/clang/Basic/DiagnosticGroups.td:429]: My initial though was to not add a flag that doesn't do anything, but I added it as you requested.

**Review by cor3ntin — APPROVED** [review]

**Pull request merged**